service/telemetry.go

```diff
@@ -224,9 +182,19 @@ func (tel *telemetryInitializer) initOpenTelemetry(attrs map[string]string, prom
 	if err != nil {
 		return fmt.Errorf("error creating otel prometheus exporter: %w", err)
 	}
+	var resAttrs []attribute.KeyValue
+	res.Attributes().Range(func(k string, v pcommon.Value) bool {
+		resAttrs = append(resAttrs, attribute.String(k, v.AsString()))
+		return true
+	})
+
+	otelRes, err := resource.New(context.Background(), resource.WithAttributes(resAttrs...))
```

Comment:
I'm wondering if we should instead use the OTel Go SDK to construct the resource then convert the SDK resource to pcommon.Resource for use in TelemetrySettings. This would let us use the SDK's resource detectors (e.g. environment variables) in addition to just the attributes specified in the Collector config.

Doing that would be an internal refactoring, so not necessarily a blocking issue here.

Reply:
I dig that idea, currently the way we get in resource attributes is using the resource detector processor in the metrics pipeline. If people are open to me making that change, I'd be happy to. cc @codeboten @bogdandrutu

Reply:
I think it's a good idea to allow users to configure telemetry for the collector to be done via SDK configuration.

Reply:
@evan-bradley @codeboten lmk if the latest change is what you're looking for!